ios/Classes/SwiftAlarmPlugin.swift

```diff
@@ -230,14 +230,13 @@ public class SwiftAlarmPlugin: NSObject, FlutterPlugin {
                 self.vibrate = false
             }
         }
-        
+
         if fadeDuration > 0.0 {
-            if volumeMax {
-                self.setVolume(volume: 1.0, enable: true)
+            if volume != nil {
+                self.setVolume(volume: volume, enable: true)
             }
-            audioPlayer.setVolume(1.0, fadeDuration: fadeDuration)
-        } else if volumeMax {
-            self.setVolume(volume: 1.0, enable: true)
+        } else if volume {
+            self.setVolume(volume: volume, enable: true)
```

Comment:
Looks to be about the same thing
        if let volumeValue = volume {
            self.setVolume(volume: volumeValue, enable: true)
        }
        if fadeDuration > 0.0 {
            audioPlayer.setVolume(1.0, fadeDuration: fadeDuration)
        }